editor/editor_log.cpp

```diff
@@ -330,7 +338,15 @@ void EditorLog::_add_log_line(LogMessage &p_message, bool p_replace_previous) {
 	}
 
 	if (p_message.type == MSG_TYPE_STD_RICH) {
-		log->append_text(p_message.text);
+		// Replace basic ANSI colors with more pleasant-looking colors, similar to what most terminals do out of the box.
+		// This also allows the printed colors to adapt to the current editor theme at the time of printing.
+		log->append_text(p_message.text
+								 .replace("color=red]", "color=" + theme_cache.red_color.to_html() + "]")
```

Comment:
```suggestion
								 .replace("[color=red]", "[color=" + theme_cache.red_color.to_html() + "]")
```

For example in `[lb]color=red]`.

Reply:
I wanted to affect the `fgcolor` and `bgcolor` tags too, without having to triple the number of replacements used :slightly_smiling_face:

Reply:
Perhaps we should add `String.replace_multiple(from: PackedStringArray, to: PackedStringArray) -> String` which will do all the replacements in one pass.

Foe this PR, I suggest something like this:

```cpp
String _replace_bbcode_colors(const String &p_string) {
    String result;
    int buffer_start = 0;
    int from = 0;
    while (true) {
        int lb_pos = p_string.find_char('[', from);
        if (lb_pos < 0) {
            break;
        }
        int rb_pos = p_string.find_char(']', lb_pos);
        if (rb_pos < 0) {
            break;
        }
        String tag = p_string.substr(lb_pos + 1, rb_pos - lb_pos - 1);
        if (tag.begins_with("color=") || tag.begins_with("bgcolor=") || tag.begins_with("fgcolor=")) {
            int eq_pos = p_string.find_char('=');
            String color = p_string.substr(eq_pos + 1, rb_pos - eq_pos - 1);
            result += p_string.substr(buffer_start, eq_pos - buffer_start + 1) + _replace_color(color);
            buffer_start = rb_pos;
        }
        from = rb_pos + 1;
    }
    result += p_string.substr(buffer_start);
    return result;
}
```

Reply:
An alternative option is to implement named color replacement on the `RichTextLabel` side. If there is a `named_color_<name>` theme property, then that color is used, otherwise the named color from `Color`.

Reply:
`replace_multiple` would be pretty useful for some use cases, but I think your custom solution should be faster.

Reply:
To me this seems logical, users will be able to use different themes for the same BBCode text. Same for fonts: it is inconvenient to write the full path to the font every time.

Reply:
I've tried my hand at implementing `String.replace_multiple()`, but didn't manage to get it to work correctly yet: https://github.com/Calinou/godot/treestring-add-replace-multiple

Reply:
I think you first need to create a [prefix tree](https://en.m.wikipedia.org/wiki/Trie) from the replacement keys, and then iterate the string character by character, checking for matches for the replacement.